### ISSUE TITLE
[Feature request] Jump over the decimal character when trying to enter a number and the integer part limit has already been attained

### ISSUE THREAD

**Issue by AlexandreBonneau** [BobKnothe/autoNumeric#317]
This is somewhat related to issue #306.

If you have `0|.00` in an input which has the `{ leadingZero: 'deny' }` option, then if you try to enter `'0'`, the caret moves to `|0.00`.

It should either stay in place `0|.00` (the current 'normal' behavior), or go to the other side of the decimal character `0.|00` (this is a new feature).

### New feature description
If you enter '2.342.423.423.423|,00 €' in an input, then you cannot enter any more numbers at that caret position since the `maximumValue` option is by default set to `'9999999999999.99'` (ie. `9.999.999.999.999,99`).

We could imagine that if you have maxed out the number of integer places available, and the caret is next to the decimal character on its left, then any other character input would just move the caret on the right side of the decimal character.
Then, if more character are entered, it would behave like it normally does, by moving the remaining decimal places to the right, as you enter new ones, like it does if you start at `2.342.423.423.423,|00 €`.

To sum up, this new feature would allow the user to continue entering numbers if the integer places are "full", by jumping over the decimal character once, then resume the normal behavior.

One thing to note; the `'autoNumeric:minExceeded'` and `'autoNumeric:maxExceeded'` events would still be sent when the user would input a character, and that character would be 'lost' ; its only purpose would be to *move* the caret over the decimal character, to allow continuing entering other numbers.

This behavior could be activated with a new option `continueOverDecimalChar` set to `true` by default.

Any thoughts on that?

**Comment by BobKnothe** [comment]
I will take this one

**Comment by BobKnothe** [comment]
added feature to  "2.0.0-beta.17" that allows the caret to jump the decimal character when just to the left of the decimal character and the user presses the decimal key or the alt decimal key.

**Comment by AlexandreBonneau** [comment]
Note: Finally we did not go with the first proposition, because that would mean a number input would be dropped in order to 'jump' over the decimal character, which would not be a decent behavior.

As Bob mentioned, in order to 'jump' to the right of the decimal character, you need to either hit de decimal character key, or the alternative decimal character key.
